macaw-aarch32/src/Data/Macaw/ARM/Semantics/TH.hs

```diff
@@ -414,79 +422,285 @@ unitValue =  M.Initial ARMDummyReg
 natReprFromIntTH :: Int -> Q Exp
 natReprFromIntTH i = [| knownNat :: M.NatRepr $(litT (numTyLit (fromIntegral i))) |]
 
+-- | A representation of an
+-- un-executed effectful generator action, where 'addr' is a generalized
+-- address (either a memory address, or a register number) and 'val' is the value
+-- to be written.
```

Comment:
I think `addr` and `val` turned into `w` - is that just pointer width now?

Reply:
Just `val`, see here: https://github.com/GaloisInc/macaw/pull/156#discussion_r466683933